sim/midas/src/main/cc/bridges/autocounter.cc

```diff
@@ -19,86 +19,117 @@ autocounter_t::autocounter_t(
     std::vector<std::string> &args,
     AUTOCOUNTERBRIDGEMODULE_struct * mmio_addrs,
     AddressMap addr_map,
+    const uint32_t event_count,
+    const char* const* event_types,
+    const uint32_t* event_widths,
+    const uint32_t* accumulator_widths,
+    const uint32_t* event_addr_hi,
+    const uint32_t* event_addr_lo,
+    const char* const* event_msgs,
+    const char* const* event_labels,
```

Comment:
Emission from Golden Gate still is heavily C oriented, and i rather not write new serialization machinery if we're going to switch to a completely new protobuf-based format soon. In the past i've struggled to include c++ constructs in the header, i think because it's being included in C compilation units with a global `-include`. 

I do think this should be changed though -- I can convert these arrays to c++ containers when the bridge driver is constructed for now.

Reply:
Thanks for the explanation.  Whatever you want to do that doesn't create a bunch of work, knowing that this will be resolved by the header reworking is helpful.